models/Document.php

```diff
@@ -571,7 +571,7 @@ public function getChildren(bool $includingUnpublished = false): Listing
     /**
      * Returns true if the document has at least one child
      */
-    public function hasChildren(bool $includingUnpublished = false): bool
+    public function hasChildren(bool $includingUnpublished = null): bool
```

Comment:
How can `null` be a correct value for a boolean?

Reply:
https://www.php.net/manual/en/language.types.declarations.php#language.types.declarations.nullable
> It is also possible to achieve nullable arguments by making null the default value. This is not recommended as if the default value is changed in a child class a type compatibility violation will be raised as the null type will need to be added to the type declaration.

Reply:
@dvesh3 shouldn't we make it true by default and not null? Like @blankse mentioned null could have side effects for child classes.

Null and true deliver the same results right?

Reply:
Yes, I know that it's technically possible. My question was about why do we *want* it to be `null` here? Why is `true` and `false` not enough, why do we need a third state? Maybe that wasn't clear enough :wink:

Reply:
@mattamon No we should make a ? sign bevor bool. null and true is not the same.

Reply:
null use the default behaviour from `Model\Document::doHideUnpublished()`
https://github.com/pimcore/pimcore/blob/64e31dba49c1a5c8ee584f2e30efc6834ce5eb16/models/Document/Dao.php#L352

Reply:
@blankse good catch, thank you. For more clarity I would use only true and false like @jdreesen mentioned. I am not sure if `hasChildren` should use the default behaviour from 'Model\Document' but I guess it makes sense if you want to check if an element has only published children if you set it in `Model\Document`

Reply:
I would not change it to `true` as it changes the default behavior like blankse shared. Moreover, we should keep it consistent with https://github.com/pimcore/pimcore/blob/c26aaed6e70b90e735951ad6d69a04698af4aec7/models/DataObject/AbstractObject.php#L386

Reply:
@mattamon It is the behaviour of 10.5, see:
https://github.com/pimcore/pimcore/blob/a4117973b09b0177f7f04ddf2e91930d5a7452cb/models/Document.php#L676
Why do you want to change it?

Reply:
@blankse got it. You are right, `null` is necessary here.